### ISSUE TITLE
Really hard to navigate due to  table width

### ISSUE THREAD

**Issue by ValWood** [pombase/canto#2037]
I think this is going to be a real problem for lots of sessions.
![too wide](https://user-images.githubusercontent.com/7359272/65895376-46e38100-e3a3-11e9-989c-bb5accafca57.jpg)

**Comment by ValWood** [comment]
I can't remember which suggestions were discussed int he past but here are some for discussion (probably again)

1 Combine organism and strain column.
2. Shorten organism name to 
M. oryzare and only keep more letters in genus if further disambiguation is required

**Comment by ValWood** [comment]
3. for alleles of type "other" which tend to be the ones with long text descriptions, only show the first few characters of the text.
Here this would be MoChai1-NSP(SPΔ(res...)

**Comment by jseager7** [comment]
It's almost inevitable that we're going to run into width issues using tables, because we can only add more properties to each concept by extending the width. About the best we can do, like you've said, is to collapse columns together (another option here is moving the PHIPO term ID into the 'Term name' column).

Unfortunately collapsing columns removes the main advantage of tables: the ability to easily compare values in categories of data. We probably need to consider which columns this comparison is and isn't useful for, and then collapse the columns where this comparison isn't so useful.

I feel like I've asked this before, but I'd be interested to know what the main reason is for keeping tables. Is it to allow easy comparison within categories of data? For example, is it useful to be able to scan down the Gene or Genotype columns? One thing that might be useful to think about is which columns you find yourself using as an 'index', i.e. to help locate data when you're looking at the table.

**Comment by ValWood** [comment]
Good points. Personally I think we need to see the table as we make annotations. I often scan to find the closest annotation to copy edit, to see if I used an allele before, to sanity check the whole contents. I think if I need to filter it would drive me crazy. It would be useful to assess though...it might be habit
@mah11  @Antonialock  @CuzickA  ?

I still that if we can make small changes to the table without loss of info  we should (until we think of a better solution).  Most tables are fine  for PomBAse Canto on a laptop screen.  However, the addition of a strain column, and longer gene IDs have pushed  it over the edge for PHI-Canto as soon as you have  something longer than normal like "allele type other"

Options 1&3 would help. I don't envisage we would ever add another column now (unless for a completely new application).

**Comment by jseager7** [comment]
> Most tables are fine for PomBAse Canto on a laptop screen. However, the addition of a strain column, and longer gene IDs have pushed it over the edge

I guess if we can keep the table narrow enough to fit on a standard laptop screen then we shouldn't need to worry about coming up with a new layout. I can look into merging some of the columns. Abbreviating the text descriptions of alleles might be trickier, but I think it should still be possible on the front-end.

**Comment by Antonialock** [comment]
I absolutely need to be able to easily see which annotations I have already created.

Some suggestions: 
- shorten organism name to make the col narrower (eg M. oryzae)
- when there is a term request, can this cell visually be "overlaid" on top of the term annotated to as a  'stopgap'?
- remove the column for term ID
-allow users to collapse columns (I guess most people will annotate on wide and large screens, but for the instances when you are finding yourself on a 11" screen then maybe it's nice to have the option to temporarily hide certain columns)

**Comment by ValWood** [comment]
> remove the column for term ID

or combine?

> allow users to collapse columns 
That would be very nice. I suspect it is hard.

**Comment by Antonialock** [comment]
> I like having the term ID in its own column. I'm aware that I'm probably unusual in the (high) extent to which I use the IDs to spot terms, but please NO to this idea.

:) for me they are just clutter since I never search on them in the annotation table (I only ever use them for term lookup at the point of creating annotations) 

Sounds like collapsible columns would be useful here - then they can viewed when needed and hidden when they are not needed

**Comment by CuzickA** [comment]
I also like the table layout to scan through the annotations.

Could the 'genotype' text be wrapped as in some of the other columns?

**Comment by jseager7** [comment]
## Response to comments

#### Shortening organism name

It looks like there are no objections to shortening the genus name, so I'll try this together with merging the strain column with the scientific name.

#### Merging Term Request column with Term Name

I think Antonia's suggestion about masking the term name column with the suggested term is interesting, but it may take a bit of work to implement. We may need some way to preserve the display of the 'stopgap' term, in an unambiguous way, in case the user is using the nearest parent term as a temporary term &ndash; that would make it easier for admins to reason about where the term should be placed in the ontology hierarchy when they create the new term... but could admins figure this out by themselves in most cases?

#### Removing the Term ID column

I think removing the term ID column is a bit too drastic &ndash; a more conservative option is to merge the Term ID column with the Term Name column, and ensure that the Term ID is always in a consistent position in the cell (preferably on its own line), so that we still preserve some ability to visually scan the data. Here's an example:

![image](https://user-images.githubusercontent.com/37659591/66306514-c4564680-e8f9-11e9-8a1a-c5d4834bb642.png)

Alternatively, if we were able to get column hiding implemented, then we could allow users to hide the Term ID column conditionally.

(@mah11 Note that it's easy enough to restrict any of these changes to only display in pathogen-host mode; in fact, this issue was prompted by a session from PHI-Canto anyway, and most of the width issues aren't so much of a problem for PomBase's instance. If I decide to try the Term ID column merge, I'll restrict it to PHI-Canto, permanently if necessary.)

#### Hiding columns

This might not be as difficult as expected; AngularJS makes it fairly easy to hide parts of the interface conditionally &ndash; though I can't say for sure since I haven't tried to hide columns like this before &ndash; I think the more difficult part might be having Canto remember the user's preferences for hidden columns after the page is refreshed or navigated away from.

Another complication is how exactly we should display a control to hide the column. Conventionally you would have this as a right-click action on the column header, but we aren't using custom right-click actions anywhere else in Canto. We could have a link to configure view settings for each table, but that might prove less intuitive to use.

#### Wrapping genotype text

This is something that's been discussed in the past (see https://github.com/pombase/canto/issues/1520#issuecomment-409909709), but I think it might have been obsoleted now by the introduction of truncating long allele names and having them expand on click.

## Mock-up of suggestions

Here's the original table row that I'm using for this example:

![image](https://user-images.githubusercontent.com/37659591/66308889-18fcc000-e900-11e9-917b-36bf43fc0494.png)

Applying most of the changes discussed above would narrow the row to the following:

![image](https://user-images.githubusercontent.com/37659591/66308908-27e37280-e900-11e9-959a-078f02342a51.png)

As the image above shows, most of the width is still be used up by the genotype name. If we decided to default to hiding the parenthetical information (expression, background, etc.), then I think the mockup-below shows the 'minimum' width we can reasonably achieve:

![image](https://user-images.githubusercontent.com/37659591/66309043-83156500-e900-11e9-963f-55ee894a4ee4.png)

It's not easy to quantify the width difference from those images, so here's a more abstract representation:

![image](https://user-images.githubusercontent.com/37659591/66310866-1ea8d480-e905-11e9-9ee8-6a9d6688cc25.png)

**Comment by ValWood** [comment]
Lets's implement 1. But would need to have a solution for if we have organisms with the same Species abbreviation.

**Comment by ValWood** [comment]
FOr thsi one

3.
for alleles of type "other" which tend to be the ones with long text descriptions, only show the first few characters of the text.
Here this would be MoChai1-NSP(SPΔ(res...)

I was suggesting that only collapse the longer free text descriptions in  allele type 'other" but keep others as they are . Rationale: we still need to see most allele type sin full because they are often  highly  similar and we need to scan the annotations for the correct one.

Does that make sense?

I also suggest  that  we move the individual changes we decide to adopt not new tickets (one for each), and close this brainstorming  ticket.

**Comment by jseager7** [comment]
> for alleles of type "other" which tend to be the ones with long text descriptions, only show the first few characters of the text. Here this would be MoChai1-NSP(SPΔ(res...)

Would you want to hide the expression as well in this case?

**Comment by ValWood** [comment]
Would you want to hide the expression as well in this case?

no, only 'contrct the description'
These are the things which push the  table width because they often have "essays" in them.

**Comment by jseager7** [comment]
@ValWood @CuzickA I'm about to open a pull request to merge the strain and organism columns in the annotation tables (on the summary page), but I'm wondering what to do about the fact that the scientific name is formatted differently in the table of metagenotype annotations.

Currently the scientific name is bold and italic &ndash; do you think it would look better without this formatting? Is it still easy enough to distinguish the genotype? 

**Before:**

![image](https://user-images.githubusercontent.com/37659591/67276006-affa7800-f4bb-11e9-903f-d11b7e4afcdb.png)

**After:**

![image](https://user-images.githubusercontent.com/37659591/67275960-99ecb780-f4bb-11e9-9068-f67efcd7af1f.png)

**Comment by CuzickA** [comment]
It would be nice to have the species name in italics. This may also need to be applied to the single species annotations too.

I'm not sure about the bold/not bold. It is easier to read when some of the text is in bold in the 'before' image above.

**Comment by ValWood** [comment]
I agree, species name in italics.
I also think single species should be the same format.
I'm fairly agnostic about  bold/non-bold. It would be easier to decide on a larger table. Let's start with bold then...

**Comment by jseager7** [comment]
> I'm fairly agnostic about bold/non-bold. It would be easier to decide on a larger table. Let's start with bold then...

The other tables (annotations, genotype, metagenotype) use non-bold for species names. I think the only reason the names are bold here is to keep them distinct from the genotype name. If you think the names are distinct enough when non-bold, then it would be more consistent to have all the scientific names non-bold.

**Comment by ValWood** [comment]
In an upcoming chat we should go through which of these are implemented, which are discarded, and which are still under consideration (these can go in new tickets to break it down a little).

I was looking for this in relation to https://github.com/pombase/canto/issues/1999

**Comment by ValWood** [comment]
Hi James, when you get chance could you summarize anything outstanding in this ticket. Then we can decide whether to implement or move on.

**Comment by jseager7** [comment]
> Hi James, when you get chance could you summarize anything outstanding in this ticket.

It looks like the unexplored solutions are as follows, mostly based on my comment above ([link](https://github.com/pombase/canto/issues/2037#issuecomment-538982834)):

* Add a control that allows the user to hide columns from the table.
* Merge the Term Request column into the Term Name column.
* Collapse allele descriptions of alleles of type 'Other' (see [comment](https://github.com/pombase/canto/issues/2037#issuecomment-539182579)), or possibly any allele type with a description.

If you want further explanation of any of these solutions, let me know and I'll go over them again.

The key changes implemented so far are the shortening of genus names, and the merging of the strain columns with the species columns.

**Comment by ValWood** [comment]
I think we can close this ticket. The solutions already implemented   seem to have improved things.

I don't think we would want to do:
Merge the Term Request column into the Term Name column.

but 
- Add a control that allows the user to hide columns from the table.
- Collapse allele descriptions of alleles of type 'Other' (see comment), or possibly any allele type with a description.

might be useful if we ever revisit this.
